packages/frontend/cypress/integration/specs/02-trade-short.spec.js

```diff
@@ -0,0 +1,277 @@
+      })
+
+      it('invalid input amount', () => {
+        cy.get('#open-short-eth-input').clear().type('\\', { delay: 200, force: true }).should('have.value', '0')
+        cy.get('#open-short-trade-details .trade-details-amount').should('contain.text', '0')
+      })
+
+      it('adjust collateral ratio', () => {
+        cy.get('.open-short-collat-ratio-input-box input')
+          .clear()
+          .type('250.', { delay: 200, force: true })
+          .should('have.value', '250.0')
+      })
+
+      it('can enter an amount into eth input, before & post trade amount match on position card and osqth input', () => {
+        // cy.get('#open-short-eth-input').clear().type('1', { force: true, delay: 200 }).should('have.value', '1')
+        cy.get('#open-short-eth-input').clear().type('8.', { force: true, delay: 200 }).should('have.value', '8.0')
+        cy.get('#open-short-eth-before-trade-balance').then((bal) => {
+          cy.get('#open-short-eth-post-trade-balance').should('contain.text', (Number(bal.text()) - 8).toFixed(2))
+        })
+
+        cy.get('#open-short-trade-details .trade-details-amount').invoke('text').should('not.equal', 0)
+
+        cy.get('#open-short-trade-details .trade-details-amount').then((val) => {
+          cy.get('#open-short-osqth-before-trade-balance').then((bal) => {
+            //post = before + input
+            cy.get('#open-short-osqth-post-trade-balance').should(
+              'contain.text',
+              new BigNumber(val.text().toString()).plus(Number(bal.text())).toFixed(2),
+            )
+          })
+          cy.get('#position-card-before-trade-balance').then((bal) => {
+            cy.get('#position-card-post-trade-balance').should(
+              'contain.text',
+              new BigNumber(val.text().toString()).plus(Number(bal.text())).toFixed(2),
+            )
+          })
+        })
+      })
+
+      it('can open short position for osqth', () => {
+        cy.get('#open-short-eth-input').clear().type('8.', { force: true, delay: 200 }).should('have.value', '8.0')
+        cy.get('#open-short-trade-details .trade-details-amount').then((v) => {
+          cy.get('#open-short-osqth-before-trade-balance').then((bal) => {
+            const inputSqth = new BigNumber(v.val().toString()).toFixed(2)
+            cy.get('#open-short-sumbit-tx-btn').then((btn) => {
+              if (btn.text().includes('Allow wrapper')) {
+                cy.get('#open-short-sumbit-tx-btn').click({ force: true })
+                trade.confirmMetamaskTransaction()
+                trade.waitForTransactionSuccess()
+              }
+              if (btn.text().includes('Deposit and sell')) {
+                cy.get('#open-short-sumbit-tx-btn').click({ force: true })
+                trade.confirmMetamaskTransaction()
+                trade.waitForTransactionSuccess()
+              }
+            })
+            cy.get('#open-short-sumbit-tx-btn').click({ force: true })
```

Comment:
@yichiehliu not sure what this line does since it seems like we already clicked the submit buttons in the preceding lines.